### ISSUE TITLE
[BUG] [go-server] openapi-generator doesn't take into account servers overrides on path level 

### ISSUE THREAD

**Issue by barzoj** [OpenAPITools/openapi-generator#1694]
##### Description

openapi-generator doesn't take into account servers overrides on path level 

##### openapi-generator version

```# openapi-generator version
3.3.3
```
##### OpenAPI declaration file content or url

test.yaml
```yaml
openapi: 3.0.0
info:
  title: title
  description: description
  version: 0.0.1

servers:
  - url: '/foo'

paths:
  /pets:
    get:
      summary: empty response
      responses:
        '200:':
          description: response ok
  /shop:
    servers:
      - url: '/bar'
    get:
      summary: empty response
      responses:
        '200':
          description: response ok
```
##### Command line used for generation

```
# openapi-generator generate -g go-server -o /api -i test.yaml -v
```

##### Current output
```
api/go/routers.go
var routes = Routes{
        {
                "Index",
                "GET",
                "/foo/",
                Index,
        },

        {
                "PetsGet",
                strings.ToUpper("Get"),
                "/foo/pets",
                PetsGet,
        },

        {
                "ShopGet",
                strings.ToUpper("Get"),
                "/foo/shop",
                ShopGet,
        },
}
```

##### Expected output 
```
var routes = Routes{
        {
                "Index",
                "GET",
                "/foo/",
                Index,
        },

        {
                "PetsGet",
                strings.ToUpper("Get"),
                "/foo/pets",
                PetsGet,
        },

        {
                "ShopGet",
                strings.ToUpper("Get"),
                "/bar/shop",
                ShopGet,
        },
}
```

**Comment by adamdecaf** [comment]
I have a similar problem: 

<details>
<summary>openapi.yaml</summary>

```yaml
openapi: "3.0.2"
info:
  version: "v1"
  title: "Test API"
  license:
    name: "Apache 2.0"
    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
servers:
  - url: https://api.moov.io/v1/
    description: Production server
tags:
  - name: Files
    description: Example

paths:
  /users/login:
    get:
      summary: Other endpoint
      operationId: checkUserLogin
      servers:
        - url: http://auth.apps.svc.cluster.local/
          description: Kubernetes Endpoint for ACH
        - url: https://api.moov.io/v1/
          description: Moov API endpoint
      responses:
        '200':
          description: success
        '403':
          description: Cookie data is invalid or expired. Login required.
  /files:
    get:
      tags:
      - Files
      summary: Gets a list of Files
      operationId: getFiles
      servers:
        - url: http://ach.apps.svc.cluster.local/
          description: Kubernetes Endpoint for ACH
        - url: https://api.moov.io/v1/ach/
          description: Moov API endpoint
      responses:
        '200':
          description: A list of File objects
          headers:
            X-Total-Count:
              description: The total number of Originators
              schema:
                type: integer
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Files'

components:
  schemas:
    File:
      properties:
        id:
          type: string
          description: File ID
          example: 3f2d23ee214
    Files:
      type: array
      items:
        $ref: '#/components/schemas/File'
```

</details>

However, `GetFiles` has the following:

```Go
func (a *FilesApiService) GetFiles(ctx context.Context) ([]File, *http.Response, error) {
	// ...
	localVarPath := a.client.cfg.BasePath + "/files"
```

I expected: 

```Go
func (a *FilesApiService) GetFiles(ctx context.Context) ([]File, *http.Response, error) {
	// ...
	localVarPath := a.client.cfg.BasePath + "/ach/files"
```

For comparison, the other endpoint works as expected: 

```Go
func (a *DefaultApiService) CheckUserLogin(ctx context.Context) (*http.Response, error) {
	// ... 
	localVarPath := a.client.cfg.BasePath + "/users/login"
```

**Comment by adamdecaf** [comment]
Also, executing these requests on the [Swagger Editor](https://editor.swagger.io/) works as expected, so this seems to be a bug in this project. Can someone point me to look somewhere to PR a fix?

**Comment by adamdecaf** [comment]
I tried my examples with `-g php` and the same unexpected `/files` path was given, so is this a problem with all codegen?

**Comment by barzoj** [comment]
Well, we will probably need to run examples against all generators to verify if it’s a generic issue or not.

**Comment by hendrytl** [comment]
It is an issue with csharp as well.

**Comment by barzoj** [comment]
it seems to be fixed in `4.0.0-beta2`, at least for `go-server`

Please verify for other generators
